File: python_files/raylib/centered_text.py
import pyray as p

SCREEN_WIDTH: int = 800
SCREEN_HEIGHT: int = 600
SCREEN_TITLE: str = "Centered Text"
SCREEN_BACKGROUND: p.Color = p.RAYWHITE


def main() -> None:
    p.init_window(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE)

    width: float = 100.0
    height: float = width
    rect: p.Rectangle = p.Rectangle(
        p.get_screen_width() / 2 - width / 2,
        p.get_screen_height() / 2 - height / 2,
        width,
        height,
    )
    color: p.Color = p.BLACK
    thickness: float = 5.0

    count: int = 0

    while not p.window_should_close():
        p.begin_drawing()
        p.clear_background(SCREEN_BACKGROUND)

        if p.is_key_pressed(p.KeyboardKey.KEY_UP):
            count += 1
        elif p.is_key_pressed(p.KeyboardKey.KEY_DOWN):
            count -= 1

        text: str = str(count)
        font_size: int = 40
        text_width: float = p.measure_text(text, font_size)
        text_x: int = int(rect.x + rect.width / 2 - text_width / 2)
        text_y: int = int(rect.y + rect.height / 2 - font_size / 2)

        p.draw_rectangle_lines_ex(rect, thickness, color)

        p.draw_text(text, text_x, text_y, font_size, color)

        p.end_drawing()

    p.close_window()


if __name__ == "__main__":
    main()

File: c_files/raylib/centered_text.c
#include "raylib.h"
#include <stdio.h>

#define TEXT_SIZE 10

int main(void)
{
  const int SCREEN_WIDTH = 800;
  const int SCREEN_HEIGHT = 600;
  const char SCREEN_TITLE[] = "Centered Text Example";
  const Color SCREEN_BACKGROUND = RAYWHITE;

  InitWindow(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE);

  float width = 150.0f;
  float height = 100.f;
  Rectangle rect =
      (Rectangle){GetScreenWidth() * 0.5f - width / 2,
                  GetScreenHeight() * 0.5f - height / 2, width, height};
  float thickness = 5.0f;
  Color rect_color = BLACK;

  int count = 0;
  char text_str[TEXT_SIZE];

  while (!WindowShouldClose())
  {
    snprintf(text_str, TEXT_SIZE, "%d", count);

    if (IsKeyPressed(KEY_UP))
    {
      count++;
    }
    else if (IsKeyPressed(KEY_DOWN))
    {
      count--;
    }

    BeginDrawing();
    ClearBackground(SCREEN_BACKGROUND);

    DrawRectangleLinesEx(rect, thickness, rect_color);

    int font_size = 40;
    float text_width = MeasureText(text_str, font_size);
    int text_x = (int)(rect.x + rect.width / 2) - text_width / 2;
    int text_y = (int)(rect.y + rect.height / 2) - font_size / 2;

    DrawText(text_str, text_x, text_y, font_size, BLACK);

    EndDrawing();
  }

  CloseWindow();

  return 0;
}